manifests/resolver.pp

```diff
@@ -122,10 +122,12 @@
     $order = "25-${defaults}-${section_name}-02"
   }
 
-  # verify accepted_payload_size is withing the allowed range per HAProxy docs
+  # verify accepted_payload_size is withing the allowed range per HAProxy docs and only set if specified
   # https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#5.3.2-accepted_payload_size
-  if ($accepted_payload_size < 512) or ($accepted_payload_size > 8192) {
-    fail('$accepted_payload_size must be atleast 512 and not more than 8192')
+  if $accepted_payload_size != undef {
```

Comment:
This entire block could be removed by instead specifying the parameter as:
`Optional[Integer[512,8192]] $accepted_payload_size = undef`